src/fedora/27/Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM fedora:27
+
+# Workaround to avoid NuGet restore timeout
+RUN dnf upgrade -y nss \
+    && dnf clean all
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
+# this does not include libraries that we need to compile different projects, we'd like
+# them in a different layer.
+RUN dnf install -y \
+        clang \
+        cmake \
+        lldb-devel \
+        llvm-devel \
+        make \
+        python \
+        which \
+    && dnf clean all
+
+# Install tools used by the VSO build automation.
+RUN dnf install -y \
+        git \
+        npm \
+        nodejs \
+        tar \
+        zip \
+    && dnf clean all
+
+RUN npm install -g azure-cli --unsafe-perm
+
+# Dependencies of CoreCLR and CoreFX.
+RUN dnf install -y \
+        glibc-locale-source \
+        libcurl-devel \
+        libgdiplus \
+        libicu-devel \
+        libunwind-devel \
+        libuuid-devel \
+        lttng-ust-devel \
+        openssl-devel \
+        uuid-devel \
+    && dnf clean all
+
+RUN dnf install -y compat-openssl10-devel --allowerasing
```

Comment:
Why not add these dependencies to one of the restores above?

Reply:
I thought it would be cleaner to have the abnormal ones (i.e. the compat installs) in their own command.

Reply:
These aren't just commands, they're layers, and you should in general use as few as you can:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

Reply:
Interesting, thanks for the link. I combined them.